### PR TITLE
Parquet support

### PR DESCRIPTION
**Pull request by willyborankin**

### CODE REVIEW

File: src/integration-test/java/io/aiven/kafka/connect/gcs/AvroParquetIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+        final Schema valueSchema =
+                SchemaBuilder.record("value")
+                        .fields()
+                        .name("name").type().stringType().noDefault()
+                        .name("value").type().stringType().noDefault()
+                        .endRecord();
+
+        final List<Future<RecordMetadata>> sendFutures = new ArrayList<>();
+        int cnt = 0;
+        for (int i = 0; i < 10; i++) {
```
Comment (HelenMel) [inline src/integration-test/java/io/aiven/kafka/connect/gcs/AvroParquetIntegrationTest.java:208]: I think there should be a generator for such a records somewhere

Comment (willyborankin) [reply src/integration-test/java/io/aiven/kafka/connect/gcs/AvroParquetIntegrationTest.java:208]: true, but so far it is not clear. the problem is that we have same tests for S3 :-(

Comment (willyborankin) [reply src/integration-test/java/io/aiven/kafka/connect/gcs/AvroParquetIntegrationTest.java:208]: I personally think that we need merge `aiven-kafka-commons`, `gcs` and `s3` connectors into one repo.

File: src/integration-test/java/io/aiven/kafka/connect/gcs/ParquetIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+        }
+    }
+
+    @Test
+    final void schemaChanged(@TempDir final Path tmpDir)
+            throws ExecutionException, InterruptedException, IOException {
+        final Map<String, String> connectorConfig = basicConnectorConfig();
+        final var compression = "none";
+        connectorConfig.put("format.output.fields", "value");
+        connectorConfig.put("format.output.fields.value.encoding", "none");
```
Comment (HelenMel) [inline src/integration-test/java/io/aiven/kafka/connect/gcs/ParquetIntegrationTest.java:308]: Lets move some general config to a function, so do not disturb test writer/reader from a main task.?

File: src/integration-test/java/io/aiven/kafka/connect/gcs/ParquetIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+        final var expectedRecords = new ArrayList<String>();
+        int cnt = 0;
+        for (int i = 0; i < 10; i++) {
+            for (int partition = 0; partition < 4; partition++) {
+                final var key = "key-" + cnt;
+                final var payload = i < 5
+                        ? "{" + "\"name\": \"user-" + cnt + "\"}"
+                        : "{" + "\"name\": \"user-" + cnt + "\", \"value\": \"value-" + cnt + "\"}";
+                final String value = i < 5
+                        ? String.format(
```
Comment (HelenMel) [inline src/integration-test/java/io/aiven/kafka/connect/gcs/ParquetIntegrationTest.java:331]: We could use ternary operator for jsonMessageSchema only. It looks like everything else stays the same

File: src/integration-test/java/io/aiven/kafka/connect/gcs/ParquetUtils.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import java.util.List;
+
+import org.apache.avro.generic.GenericRecord;
+import org.apache.parquet.avro.AvroParquetReader;
+import org.apache.parquet.io.DelegatingSeekableInputStream;
+import org.apache.parquet.io.InputFile;
+import org.apache.parquet.io.SeekableInputStream;
+import org.testcontainers.shaded.org.apache.commons.io.FileUtils;
+
+class ParquetUtils {
```
Comment (HelenMel) [inline src/integration-test/java/io/aiven/kafka/connect/gcs/ParquetUtils.java:33]: It doesn't look like something specific to GCS. Could we move it to commons?

File: src/main/java/io/aiven/kafka/connect/gcs/GcsSinkTask.java [hunk 5]
```diff
@@ -121,40 +100,20 @@ public void flush(final Map<TopicPartition, OffsetAndMetadata> currentOffsets) {
...
-                        outputWriter.writeRecord(record);
-                    }
-                }
-            }
-            storage.create(blob, baos.toByteArray());
+                .newBuilder(config.getBucketName(), config.getPrefix() + filename)
+                .build();
+        try (final var out = Channels.newOutputStream(storage.writer(blob));
+             final var writer = OutputWriter.builder()
+                     .withExternalProperties(config.originalsStrings())
```
Comment (HelenMel) [inline src/main/java/io/aiven/kafka/connect/gcs/GcsSinkTask.java:107]: Sorry, maybe I missed something. What does exactly originalStrings are responsible for?

**Comment by HelenMel** [comment]
Couple of comments

File: src/integration-test/java/io/aiven/kafka/connect/gcs/ParquetUtils.java (hunk 4 above)
Comment (willyborankin) [reply src/integration-test/java/io/aiven/kafka/connect/gcs/ParquetUtils.java:33]: we can't move it to commons, this class is for testing pupose only. so lets keep is as is, and discuss how can we improve our common tests part

File: src/main/java/io/aiven/kafka/connect/gcs/GcsSinkTask.java (hunk 5 above)
Comment (willyborankin) [reply src/main/java/io/aiven/kafka/connect/gcs/GcsSinkTask.java:107]: by default config is `Map<String, Object>`, but props are passed as strings, so `originalsStrings` returns `Map<String, String>`

File: src/integration-test/java/io/aiven/kafka/connect/gcs/ParquetIntegrationTest.java (hunk 3 above)
Comment (willyborankin) [reply src/integration-test/java/io/aiven/kafka/connect/gcs/ParquetIntegrationTest.java:331]: better `if/else` as result it is readble

**Comment by willyborankin** [comment]
@HelenMel last commit contains comments fix

**Review by HelenMel — APPROVED** [review]

**Pull request merged**